src/physics/forceCal/forceCal.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "physics/forceCal/forceCal.h"
+#include "utils/ArrayUtils.h"
+#include <unordered_map>
+#include <vector>
+
+void force_stroemer_verlet(const Simulation& sim)
+{
+    for (auto& p1 : sim.container) {
+        std::array<double, 3> f_i = { 0.0, 0.0, 0.0 };
+        for (auto& p2 : sim.container) {
+            double m_mul = p1.getM() * p2.getM();
+            double dist = std::pow(ArrayUtils::L2Norm(p1.getX() - p2.getX()), 3);
```

Comment:
currently you go over all pair of particles here and check for distance, but you can think of ways (for loops - index bases for eg.) where you simply never come across this case.